### PR TITLE
Fix the path block won't decay to dirt when you place a solid block above it (Fix #904)

### LINKED ISSUE #904: The path block won't decay to dirt when you place a solid block above it
# 🐞 I found a bug

### 📸 Screenshots / Videos

image

### ▶ Steps to Reproduce

1. Place a dirt block on top of a path block

### ✔ Expected Behavior

The path block should decay to dirt

### ❌ Actual Behavior

The path block remains as a path block forever

### 📋 Debug information

* PowerNukkit version: 1.4.0.0-PN-SNAPSHOT at e2149a5
* Debug link: https://debugpaste.powernukkit.org/pastes/alavehicuh

### PR DESCRIPTION
**Pull request by GoodLucky777**
Fix the path block won't decay to dirt when you place a solid block above it (Fix #904)

Dirt Path:
  - Fix the path block won't decay to dirt when you place a solid block above it
  - Fix hardness and resistance

Grass Block:
  - Fix resistance

### CODE REVIEW

File: src/main/java/cn/nukkit/block/BlockGrassPath.java [hunk 1]
```diff
@@ -59,6 +64,13 @@ public boolean canSilkTouch() {
 
     @Override
     public int onUpdate(int type) {
+        if (type == Level.BLOCK_UPDATE_RANDOM) {
+            if (this.up().isSolid()) {
+                this.level.setBlock(this, Block.get(BlockID.DIRT), false, true);
+            }
+            
+            return Level.BLOCK_UPDATE_RANDOM;
+        }
```
Comment (joserobjr) [inline src/main/java/cn/nukkit/block/BlockGrassPath.java:73]: According to single-player tests, It should decay immediately after a solid block is placed on top of it, so it should be `BLOCK_UPDATE_NORMAL`

**Review by joserobjr — CHANGES_REQUESTED** [review]
Thank you for the PR.
I only found one issue with it, grass path blocks don't receive random ticks, they should update on normal block updates

File: src/main/java/cn/nukkit/block/BlockGrassPath.java (hunk 1 above)
Comment (GoodLucky777) [reply src/main/java/cn/nukkit/block/BlockGrassPath.java]: I think I accidentally write it, I edited to BLOCK_UPDATE_NORMAL

**Review by joserobjr — APPROVED** [review]
Thank you!

**Pull request merged**